client/templates/view_data/view_data.js

```diff
@@ -331,22 +331,29 @@ function classStats(envId, sParams, obsId) {
     text: "Classroom Summary"
   }).appendTo(classRoomSummary);
 
-  sParams.map(function(param) {
+  sParams.map(function(param, idx) {
     var newObject = {};
+    var paramWithData = [];
     for (con in filteredResults) {
       var next = filteredResults[con]['info'];
       studTrack.add(next['studentId']);
       if (next[param]) {
         if (next[param] in newObject) {
           newObject[next[param]] += 1;
         } else {
+          paramWithData.push(next[param]);
           newObject[next[param]] = 1;
         }
       }
     }
+
+    var sequenceParameters = SequenceParameters.find({'children.envId':envId}).fetch()[0];
+    var paramPosition = `parameter${idx}`;
+    var parameters = sequenceParameters.children[paramPosition].split(",");
+    parameters.filter((obj) => { return paramWithData.indexOf(obj) == -1; }).map((item) => { newObject[item] = 0; })
```

Comment:
Instead of `.map`, should we be using `.forEach`?

Reply:
yeah good call. No need to store in memory if not being used.